src/target/riscv/batch.h

```diff
@@ -78,4 +78,9 @@ size_t riscv_batch_available_scans(struct riscv_batch *batch);
 /* Return true iff the last scan in the batch returned DMI_OP_BUSY. */
 bool riscv_batch_dmi_busy_encountered(const struct riscv_batch *batch);
 
+/* TODO: The function is defined in `riscv-013.c`. This is done to reduce the
+ * diff of the commit. The intention is to move the function defenition to
+ * `batch.c` in a separate commit. */
```

Comment:
> The intention is to move the function definition to batch.c

I would prefer to only keep batch-related code in `batch.c`. Since the code to decode JTAG transactions is generic and not strictly related to batches, putting it to batch.c does not seem to be correct.

Instead, it could go to a new module, e.g. `riscv013-jtag-dtm.c/h`, `riscv013-logging.c/h`, `riscv013-utils.c/h`, depending on your preference.

Reply:
Addressed.